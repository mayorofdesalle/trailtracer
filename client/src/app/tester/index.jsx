import '@testing-library/jest-dom';

import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import './mocks/i18n.mock';
import './mocks/matchMedia.mock';
import './mocks/resizeObserver.mock';

import store from '@app/store';
import GlobalStyle from '@components/style/GlobalStyle';
import Theme from '@components/style/Theme';

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                <GlobalStyle />
                <BrowserRouter>
                    {children}
                </BrowserRouter>
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };
