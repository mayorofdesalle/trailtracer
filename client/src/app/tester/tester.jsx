import { render } from '@testing-library/react'
import { Provider } from 'react-redux';

import './mocks/matchMedia.mock';
import store from '@app/store';
import Theme from '@components/style/Theme'

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                {children}
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };