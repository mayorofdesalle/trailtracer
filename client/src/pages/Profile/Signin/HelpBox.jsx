import { useTranslation } from 'react-i18next';
import styled, { useTheme } from 'styled-components';

import { BentoBox } from '@components/ui/bento';
import { Button } from '@components/ui';
import { Icon } from '@components/ui';
import { Text } from '@components/ui';

const HelpBoxInner = styled(BentoBox)`
    grid-area: HelpBox;
    padding: 8%;
    flex-direction: column;
    justify-content: space-around;

    & > svg {
        height: 4dvh;
        width: 4dvh;
    }
`;

const HelpBox = (props) => {
    const theme = useTheme();
    const { t } = useTranslation();

    return (
        <HelpBoxInner as={Button} $glass {...props}>
            <Icon name='login-box-fill' color={theme.colors.primary} />
            <Text $color={theme.colors.text} $heading $ratio={0.9}>{t('signinPage.help')}</Text>
        </HelpBoxInner>
    );
};

export default HelpBox;