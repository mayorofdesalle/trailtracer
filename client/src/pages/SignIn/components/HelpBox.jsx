import styled from 'styled-components';

import BentoBox from '@components/ui/Bento/Box';

const HelpBox = styled(BentoBox)`
    grid-area: HelpBox;
`;

export default HelpBox;