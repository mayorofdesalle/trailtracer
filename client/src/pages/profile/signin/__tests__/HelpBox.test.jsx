import { describe, expect, it, render, screen } from '@app/tester';

import HelpBox from '../HelpBox';

describe('HelpBox', () => {

    it('should render correctly', () => {
        render(<HelpBox />);
        const helpBox = screen.getByRole('button');
        expect(helpBox).toBeInTheDocument();
        expect(helpBox).toHaveStyle({ gridArea: 'HelpBox' });

        expect(helpBox).toHaveTextContent(/help/i);

        const icon = screen.getByRole('img');
        expect(icon).toBeInTheDocument();
    });

});