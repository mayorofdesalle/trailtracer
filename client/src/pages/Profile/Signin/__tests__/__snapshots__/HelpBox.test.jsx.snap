// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HelpBox > should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-hLQSwg jXhJHN sc-beySPh sc-guDLey sc-irLvIq jVdZA cVargU ddLmsq"
    type="button"
  >
    <svg
      class="sc-jTQCzO ldKJNf"
      fill="#435049"
      role="img"
    >
      <title>
        login-box-fill
      </title>
      <use
        href="http://localhost:3000/src/assets/icons.svg#login-box-fill"
      />
    </svg>
    <div
      class="sc-iBdnpw hKJUwH"
    >
      signinPage.help
    </div>
  </button>
</DocumentFragment>
`;
